build/ModifiedFile.cs

```diff
@@ -0,0 +1,38 @@
+﻿// ReSharper disable RedundantUsingDirective
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using Nuke.Common;
+using Nuke.Common.IO;
+
+public class ModifiedFile : IDisposable
```

Comment:
Maybe? (not 100% happy with it myself)
```suggestion
public class RestoreOriginalFileContentsOnDispose : IDisposable
```